LoRaEngine/LoraKeysManagerFacade/LoRaDevAddrCacheFunction.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using LoRaTools;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Azure.WebJobs.Extensions.Http;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    internal class LoRaDevAddrCacheFunction
+    {
+        private readonly LoRaDevAddrCache loRaDevAddrCache;
+        private readonly ILogger<LoRaDevAddrCacheFunction> logger;
+
+        public LoRaDevAddrCacheFunction(LoRaDevAddrCache loRaDevAddrCache, ILogger<LoRaDevAddrCacheFunction> logger)
+        {
+            this.loRaDevAddrCache = loRaDevAddrCache;
+            this.logger = logger;
+        }
+
+        [FunctionName("StoreInDevAddrCache")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "post", Route = "storeindevaddrcache")] HttpRequest req)
+        {
+            try
+            {
+                VersionValidator.Validate(req);
+            }
+            catch (IncompatibleVersionException ex)
+            {
+                return new BadRequestObjectResult(ex.Message);
+            }
+
+            var requestBody = await req.ReadAsStringAsync();
+            if (string.IsNullOrEmpty(requestBody))
+            {
+                return new BadRequestObjectResult("missing body");
+            }
+
+            var devAddrCacheInfo = JsonConvert.DeserializeObject<DevAddrCacheInfo>(requestBody);
```

Comment:
I'm contemplating if leaking a server specific type (cache info) through the API is a good idea. Usually keeping them separate is a good idea. We could also use the endpoint for something more generic, if we name it something like "DeviceJoinNotification", we could hook in other things into the join process from there. WDYT?

Reply:
I was thinking about making it more generic. DevAddrCacheInfo is now extending IoTHubDeviceInfo class (which is already shared, see #1777), by only adding LastUpdatedTwins.

We could ideally have a more generic DeviceJoinNotification function that is actually only taking IoTHubDeviceInfo as input (as anyways we are not needing the LastUpdatedTwins for now).

Would it be better?

Reply:
Not sure we can consolidate the 2, looks like the required information on the 2 is quite different. The one from the facade only has DevEUI, DevAddr and PrimaryKey. Whereas the one in the LNS also has the GatewayId and the NwkSKey.

Reply:
Followed your suggestion with commits f93500a669b418694746f7eb3a9f5cbb5186c4f0 and 30302f608e1a16b5e3ff25600ac460639bfd9a97